src/parser.ts

```diff
@@ -1,132 +1,83 @@
-            currentLine.includes(singlelineCardSeparator)
-        ) {
-            cardType = lines[i].includes(singlelineReversedCardSeparator)
-                ? CardType.SingleLineReversed
-                : CardType.SingleLineBasic;
-            cardText = lines[i];
-            firstLineNo = i;
-            if (i + 1 < lines.length && lines[i + 1].startsWith("<!--SR:")) {
-                cardText += "\n" + lines[i + 1];
-                i++;
-            }
-            lastLineNo = i;
-            cards.push(new ParsedQuestionInfo(cardType, cardText, firstLineNo, lastLineNo));
-            cardType = null;
-            cardText = "";
-        } else if (
-            cardType === null &&
-            ((convertHighlightsToClozes && /==.*?==/gm.test(currentLine)) ||
-                (convertBoldTextToClozes && /\*\*.*?\*\*/gm.test(currentLine)) ||
-                (convertCurlyBracketsToClozes && /{{.*?}}/gm.test(currentLine)))
-        ) {
-            cardType = CardType.Cloze;
-
-            // Explicitly don't change firstLineNo, as we might not see the cloze markers on the first line
-            // of a multi line cloze question. I.e. firstLineNo may be less than i;
-        } else if (currentLine.trim() === multilineCardSeparator) {
-            cardType = CardType.MultiLineBasic;
-            // Explicitly don't change firstLineNo, as per above comment
-        } else if (currentLine.trim() === multilineReversedCardSeparator) {
-            cardType = CardType.MultiLineReversed;
-            // Explicitly don't change firstLineNo, as per above comment
-        } else if (currentLine.startsWith("```") || currentLine.startsWith("~~~")) {
-            const codeBlockClose = currentLine.match(/`+|~+/)[0];
-            while (i + 1 < lines.length && !lines[i + 1].startsWith(codeBlockClose)) {
-                i++;
-                cardText += "\n" + lines[i];
-            }
-            cardText += "\n" + codeBlockClose;
-            i++;
+    if(parser === undefined) {
+        // if parser is not provided explicitly, use the parser configured
+        // with the plugin settings provided by the user
+        if(defaultParser === null) {
+            if(defaultSettings === null) throw Error("Something went wrong. The variable 'defaultSettings' was not initialized yet.");
+            defaultParser = generateParser(defaultSettings);
         }
+        parser = defaultParser;    
     }
 
-    if (cardType && cardText) {
-        lastLineNo = lines.length - 1;
-        cards.push(new ParsedQuestionInfo(cardType, cardText, firstLineNo, lastLineNo));
-    }
+    // Use this function when you call the parse method
+	try {
+		cards = parser.parse(text + "\n\n\n",  {
+			CardType,
+			createParsedQuestionInfo: (cardType: CardType, text: string, firstLineNum: number, lastLineNum: number) => {
+				return new ParsedQuestionInfo(cardType, text, firstLineNum, lastLineNum);
+			}
+		});
+	} catch (error) {
+		if (error instanceof SyntaxError) {
```

Comment:
Why did you add the SyntaxError here where could this come from?

Reply:
> Why did you add the SyntaxError here where could this come from? Where and when gets this thrown?

You are right. There is no chance that this SyntaxError is generated because we wrote robust parsing rules. Whatever the user types in, the parser will produce a valid end state.

Please feel free to remove it. It is really never triggered.

Reply:
ok just got your messages displayed jet I will remove the special case is easier for testing if we can't trigger an syntax error jet